examples/ssd/train.py

```diff
@@ -0,0 +1,179 @@
+        label = label[param['index']]
+
+        # 4. Resizing with random interpolatation
+        _, H, W = img.shape
+        img = resize_with_random_interpolation(img, (self.size, self.size))
+        bbox = transforms.resize_bbox(bbox, (H, W), (self.size, self.size))
+
+        # 5. Random horizontal flipping
+        img, params = transforms.random_flip(
+            img, x_random=True, return_param=True)
+        bbox = transforms.flip_bbox(
+            bbox, (self.size, self.size), x_flip=params['x_flip'])
+
+        # Preparation for SSD network
+        img -= self.mean
+        mb_loc, mb_label = self.coder.encode(bbox, label)
+
+        return img, mb_loc, mb_label
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--batchsize', type=int, default=32)
+    parser.add_argument('--gpu', type=int, default=-1)
+    parser.add_argument('--out', default='result')
+    parser.add_argument('--resume')
+    args = parser.parse_args()
+
+    model = SSD300(
+        n_fg_class=len(voc_detection_label_names),
+        pretrained_model='imagenet')
+    model.use_preset('evaluate')
+    train_chain = MultiboxTrainChain(model)
+    if args.gpu >= 0:
+        chainer.cuda.get_device(args.gpu).use()
+        model.to_gpu()
+
+    train = TransformDataset(
+        ConcatenatedDataset(
+            VOCDetectionDataset(year='2007', split='trainval'),
+            VOCDetectionDataset(year='2012', split='trainval')
+        ),
+        Transform(model.coder, model.insize, model.mean))
+    train_iter = chainer.iterators.MultiprocessIterator(
+        train, args.batchsize, n_processes=2)
+
+    test = VOCDetectionDataset(
+        year='2007', split='test',
+        use_difficult=True, return_difficult=True)
+    test_iter = chainer.iterators.SerialIterator(
+        test, args.batchsize, repeat=False, shuffle=False)
+
+    # initial lr is set to 1e-3 by ExponentialShift
+    optimizer = chainer.optimizers.MomentumSGD()
+    optimizer.setup(train_chain)
+    for param in train_chain.params():
+        if param.name == 'b':
+            param.update_rule.add_hook(GradientScaling(2))
+        else:
+            param.update_rule.add_hook(WeightDecay(0.0005))
+
+    updater = training.StandardUpdater(train_iter, optimizer, device=args.gpu)
+    trainer = training.Trainer(updater, (120000, 'iteration'), args.out)
+    trainer.extend(
+        extensions.ExponentialShift('lr', 0.1, init=1e-3),
+        trigger=triggers.ManualScheduleTrigger([80000, 100000], 'iteration'))
+
+    trainer.extend(
+        DetectionVOCEvaluator(
+            test_iter, model, use_07_metric=True,
+            label_names=voc_detection_label_names),
+        trigger=(10000, 'iteration'))
+
+    log_interval = 10, 'iteration'
+    trainer.extend(extensions.LogReport(trigger=log_interval))
+    trainer.extend(extensions.observe_lr(), trigger=log_interval)
+    trainer.extend(extensions.PrintReport(
+        [
```

Comment:
It works, but we can make it look better by stop using an extra line.
https://github.com/chainer/chainer/blob/master/examples/mnist/train_mnist.py#L115